### PR TITLE
APB-5080 [GW] IF connector

### PR DESCRIPTION
**Pull request by geoffreywatson**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala [hunk 1]
```diff
@@ -191,21 +183,20 @@ class DesConnector @Inject()(httpClient: HttpClient, metrics: Metrics, agentCach
...
+      getWithDesHeaders(s"GetInactiveRelationships", url, desAuthToken, desEnv).map { response =>
+        response.status match {
+          case Status.OK =>
+            response.json.as[InactiveRelationshipResponse].relationship.filter(isNotActive)
+          case Status.BAD_REQUEST                             => Seq.empty
+          case Status.NOT_FOUND                               => Seq.empty
+          case _ if response.body.contains("AGENT_SUSPENDED") => Seq.empty
+          case other: Int =>
+            throw UpstreamErrorResponse(response.body, other, other)
+        }
```
Comment (barrysims) [inline app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala:199]: The three match blocks (lines 120, 141, 190) could all be factored out into one.

File: app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+      getWithIFHeaders(s"GetInactiveRelationships", url, ifAuthToken, ifEnv).map { response =>
+        response.status match {
+          case Status.OK =>
+            response.json.as[InactiveRelationshipResponse].relationship.filter(isNotActive)
+          case Status.BAD_REQUEST                             => Seq.empty
+          case Status.NOT_FOUND                               => Seq.empty
+          case _ if response.body.contains("AGENT_SUSPENDED") => Seq.empty
+          case other: Int =>
+            throw UpstreamErrorResponse(response.body, other, other)
+        }
```
Comment (barrysims) [inline app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala:140]: and these

**Review by barrysims — APPROVED** [review]

File: app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala (hunk 1 above)
Comment (geoffreywatson) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala:199]: yeah...these have always been this way...wanted to keep existing untouched for time being...once we get things working we can look at code refactoring

File: app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala (hunk 2 above)
Comment (geoffreywatson) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala:140]: using same pattern as DES but good that you spotted this

Comment (barrysims) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala:140]: :+1:

File: app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala (hunk 1 above)
Comment (barrysims) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala:199]: :+1:

**Pull request merged**